internal/providers/aws/signing_v4.go

```diff
@@ -74,6 +78,42 @@ func sortKeys(strMap map[string][]string) []string {
 	return keys
 }
 
+// SignRequest modifies an http.Request to include an AWS V4 signature based on the provided credentials.
+func SignRequest(req *http.Request, service string, creds Credentials, theTime time.Time, sigVersion string) error {
```

Comment:
Nit: `SignRequest` -> `SignRequestV4`

Reply:
SignRequest calls SignV4a or SignV4 depending on the passed `sigVersion`.
It just happens to be defined in the `signing_v4.go` file but I can also move it to it's own file.
Perhaps it doesn't even make sense to separate both implementations into two files. Especially not if they expose a shared entrypoint.